Source/Diagnostics/ReducedDiags/DifferentialLuminosity.cpp

```diff
@@ -202,83 +199,111 @@ void DifferentialLuminosity::ComputeDiags (int step)
+                    index_type const j_2 = indices_2[i_2];
+
+                    Real p1t=0, p1x=0, p1y=0, p1z=0; // components of 4-momentum of particle 1
+                    Real const u1_sq =  u1x[j_1]*u1x[j_1] + u1y[j_1]*u1y[j_1] + u1z[j_1]*u1z[j_1];
+                    if (species1_is_photon) {
+                        // photon case (momentum is normalized by m_e in WarpX)
+                        p1t = PhysConst::m_e*std::sqrt( u1_sq );
+                        p1x = PhysConst::m_e*u1x[j_1];
+                        p1y = PhysConst::m_e*u1y[j_1];
+                        p1z = PhysConst::m_e*u1z[j_1];
+                    } else {
+                        p1t = m1*std::sqrt( c_sq + u1_sq );
+                        p1x = m1*u1x[j_1];
+                        p1y = m1*u1y[j_1];
+                        p1z = m1*u1z[j_1];
+                    }
+
+                    Real p2t=0, p2x=0, p2y=0, p2z=0; // components of 4-momentum of particle 2
+                    Real const u2_sq =  u2x[j_2]*u2x[j_2] + u2y[j_2]*u2y[j_2] + u2z[j_2]*u2z[j_2];
+                    if (species2_is_photon) {
+                        // photon case (momentum is normalized by m_e in WarpX)
+                        p2t = PhysConst::m_e*std::sqrt(u2_sq);
+                        p2x = PhysConst::m_e*u2x[j_2];
+                        p2y = PhysConst::m_e*u2y[j_2];
+                        p2z = PhysConst::m_e*u2z[j_2];
+                    } else {
+                        p2t = m2*std::sqrt( c_sq + u2_sq );
+                        p2x = m2*u2x[j_2];
+                        p2y = m2*u2y[j_2];
+                        p2z = m2*u2z[j_2];
+                    }
+
+                    // center of mass energy in eV
+                    Real const E_com = c_over_qe * std::sqrt(m1*m1*c_sq + m2*m2*c_sq + 2*(p1t*p2t - p1x*p2x - p1y*p2y - p1z*p2z));
+
+                    // determine particle bin
+                    int const bin = int(Math::floor((E_com-bin_min)/bin_size));
+
+                    if ( bin<0 || bin>=num_bins ) { continue; } // discard if out-of-range
+
+                    Real const inv_p1t = 1.0_rt/p1t;
+                    Real const inv_p2t = 1.0_rt/p2t;
+
+                    Real const beta1_sq = (p1x*p1x + p1y*p1y + p1z*p1z) * inv_p1t*inv_p1t;
+                    Real const beta2_sq = (p2x*p2x + p2y*p2y + p2z*p2z) * inv_p2t*inv_p2t;
+                    Real const beta1_dot_beta2 = (p1x*p2x + p1y*p2y + p1z*p2z) * inv_p1t*inv_p2t;
+
+                    // Here we use the fact that:
+                    // (v1 - v2)^2 = v1^2 + v2^2 - 2 v1.v2
+                    // and (v1 x v2)^2 = v1^2 v2^2 - (v1.v2)^2
+                    // we also use beta=v/c instead of v
+
+                    Real const radicand = beta1_sq + beta2_sq - 2*beta1_dot_beta2 - beta1_sq*beta2_sq + beta1_dot_beta2*beta1_dot_beta2;
+
+                    Real const dL_dEcom = PhysConst::c * std::sqrt( radicand ) * min_N * w1[j_1] * w2[j_2] / dV / bin_size * dt; // m^-2 eV^-1
```

Comment:
Could you add a comment to explain `min_N`:
```
Scale the number of collisions by multiplying by `min_N`
to reflect the fact that we only sampled `max_N` pairs instead of `NI1*NI2`
```